JavaSource/org/unitime/timetable/model/Class_.java

```diff
@@ -1219,6 +1219,7 @@ public Object clone(){
 		newClass.setSchedulingSubpart(getSchedulingSubpart());
 		newClass.setCancelled(isCancelled());
 		newClass.setLms(getLms());
+		newClass.setFundingDept(getFundingDept()); //TODO Effective?
```

Comment:
This would not work -- it is needed to lookup the funding department in the new academic session, not link the one from the previous session. See how it is done for subject areas [here](https://github.com/UniTime/unitime/blob/dev_funding_dept/JavaSource/org/unitime/timetable/util/SessionRollForward.java#L993-L996).
```
if (fromSubjectArea.getFundingDept() != null) {
    Department toFundingDept = fromSubjectArea.getFundingDept().findSameDepartmentInSession(toSession);
    toSubjectArea.setFundingDept(toFundingDept);
}
```